apps/web/src/pages/Teams/TeamMembersPage/components/EditTeamMemberDialog.tsx

```diff
@@ -0,0 +1,210 @@
+              defaultMessage: "Member roles updated successfully",
+              id: "ALIgEC",
+              description:
+                "Alert displayed to user when a team member's roles have been updated",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Member role update failed",
+            id: "Ly2bBb",
+            description:
+              "Alert displayed to user when an error occurs while editing a team member's roles",
+          }),
+        );
+      });
+  };
+
+  const roleOptions = getTeamBasedRoleOptions(availableRoles, intl);
+
+  const userName = getFullNameLabel(user.firstName, user.lastName, intl);
+
+  const label = intl.formatMessage(
+    {
+      defaultMessage: "Edit membership for {userName}",
+      id: "oVgVhK",
+      description: "Label for the form to edit a users team membership",
+    },
+    {
+      userName,
+    },
+  );
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <Button color="primary" mode="outline" aria-label={label}>
+          <PencilIcon data-h2-height="base(x.75)" data-h2-width="base(x.75)" />
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>{label}</Dialog.Header>
+        <Dialog.Body>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage({
+              defaultMessage: "Change or remove this membership's permissions.",
+              id: "qPxrKm",
+              description: "Help text for the edit team membership form",
+            })}
+          </p>
+          <FormProvider {...methods}>
+            <form onSubmit={handleSubmit(handleSave)}>
+              {/** Note: Only one option since we are editing this user */}
+              <input type="hidden" name="user" value={user.id} />
+              <Select
+                id="userDisplay"
+                name="userDisplay"
+                disabled
+                label={intl.formatMessage({
+                  defaultMessage: "Manager name",
+                  id: "1BEtoY",
+                  description:
+                    "Label for the user select field on team membership form",
+                })}
+                options={[
+                  {
+                    value: user.id,
+                    label: userName,
+                  },
+                ]}
+              />
+              {/** Note: Only one option since we are editing this team's users */}
+              <input type="hidden" name="team" value={team.id} />
+              <Select
+                id="teamDisplay"
+                name="teamDisplay"
+                disabled
+                label={intl.formatMessage({
+                  defaultMessage: "Team",
+                  id: "0AaeXe",
+                  description:
+                    "Label for the team select field on team membership form",
+                })}
```

Comment:
And here

Reply:
Hidden in [hide optional label for disabled inputs](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5811/commits/ebef183a812be368ef42022f7f331ddc28113121)